### ISSUE TITLE
[Bug]: Nixpack wrong exposed port when previously changed and with static site enabled

### ISSUE THREAD

**Issue by Yarmeli** [coollabsio/coolify#1523]
### Description

I found this issue when trying to deploy a website using Astro.

I set the `Is Static Site` option to `False` when I initially imported my project from Github, which caused a few too many issues.

In summary, when the `Is Static Site` option is set to `True`, the Exposed port input is disabled and any previous value that was saved there before will be used, instead of the `default port 80`

I believe that if `is static site` is enabled, then the exposed port should either:
- previous values should be ignored and it should use port 80 when deploying
- automatically be set to 80 in the front-end
  - this can be annoying if you toggle the button and end up having to reset your port value to what it was previously
  - maybe toggle between 80 when enabled and whatever was there before when disabled?

### Minimal Reproduction (if possible, example repository)

Example of an Astro project: `https://github.com/satnaing/astro-paper`

- Create new application, import Astro project from a public GitHub
- Run using Nixpack using default provided settings (i.e. with Static Mode set to `FALSE`, port to 3000)
  - Container complains that application cannot be reached on localhost:3000
  - Logs will show application is running on localhost:4321 (i.e. default astro dev port)
- Set the default exposed port to `4321`
- Redeploy the application
  - Container shows as healthy
  - Cannot reach the application on `domain.xyz` or `domain.xyz:4321`
    - Returns `Bad Gateway` error
    - I think this is intended behaviour since I don't think we want apps being deployed in dev mode
    Could you confirm this? @andrasbacsai 
- Change Nixpack to use Static mode and set `/dist` as publish directory
- Redeploy changes
  - Container is not healthy since it is still looking at `localhost:4321`

To fix this issue:
- set Static Mode to `FALSE`
- Update port to `80` and save.
- set Static Mode to `TRUE` and publish directory to `/dist`
- <ins>**FORCE REDEPLOY**</ins> <-- The cached version will still deploy the application in dev mode (i.e. with `npm run start`)

### Version

v4.0.0-beta.153

**Comment by andrasbacsai** [comment]
tldr: You need to disable `static build`, set port to 80, enable `static build` and set the publish dir to `/dist` - just as you wrote.

In the next release, I will fix this. Whenever you change to `static build`, the configuration will be updated (exposes ports, traefik labeling, and flag the next deployment to not use the locally available image, as the configuration changed) and everything will work.

**Comment by jclapadula** [comment]
Leaving this for future generations:

Found this bug because the deployment of an Astro site with SSR was not working, I ended up with this configuration in my astro.config.msj:
`{...
  server: { port: 3000, host: true }
...}`

Of course, in coolify you also have to configure the exposed port 3000, and it works!
Must say: at the end I don't know if only host is needed or also port.

**Comment by Yarmeli** [comment]
> Leaving this for future generations:
> 
> Found this bug because the deployment of an Astro site with SSR was not working, I ended up with this configuration in my astro.config.msj: `{... server: { port: 3000, host: true } ...}`
> 
> Of course, in coolify you also have to configure the exposed port 3000, and it works! Must say: at the end I don't know if only host is needed or also port.

For SSR, I believe you need both. - I tried with host only, port only and both, and it only worked when both values were present.

I ended up adding the host and server in the `package.json` instead:

`"start": "HOST=0.0.0.0 PORT=4321 node ./dist/server/entry.mjs"`

In Coolify, I set the exposed port to `4321` and whenever the application is deployed, the nixpack build flow (i.e. `pnpm install`, `pnpm build`, `pnpm start`) ends up running with the extra commands
